### PR TITLE
Added support for labels and existing PR check

### PR DESCRIPTION
**Pull request by vitalyal**
Hello,
Thanks for publishing your action. We want to use it in our flow and faced a need to add another step next to it not to fail in case PR already exists. 
So here is a PR not to fail and report via output if PR was created or not. Also support for labels is added.

Note: i'm not sure how to easily do integration testing for these features so no testing performed yet

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -19,21 +19,42 @@ async function run(): Promise<void> {
...
-
-    if (reviewers.length > 0) {
-      await octokit.pulls.createReviewRequest({
-        owner: pullParams.owner,
-        repo: pullParams.repo,
-        pull_number: pullNumber,
-        reviewers,
-      });
-    }
+    try {
```
Comment (thomaseizinger) [inline src/index.ts:22]: Do we have to do nested `try`s? I think there should be a way of having only a single one.

File: src/index.ts [hunk 2]
```diff
@@ -19,21 +19,42 @@ async function run(): Promise<void> {
...
+          issue_number: pullNumber,
+          labels: labels,
+        });
+      }
 
-    setOutput('number', pullNumber.toString());
-    setOutput('html_url', htmlUrl);
+      setOutput('number', pullNumber.toString());
+      setOutput('html_url', htmlUrl);
+      setOutput('pr_created', 'true');
```
Comment (thomaseizinger) [inline src/index.ts:47]: Can we name this one just `created`? We are in the context of a PR here anyway so the naming doesn't need to repeat that.

**Review by thomaseizinger — CHANGES_REQUESTED** [review]
I am happy to merge these features, I think they make sense to have. Esp. idempotency is a nice one.

> Note: i'm not sure how to easily do integration testing for these features so no testing performed yet

You can do that pretty easily by opening a PR on your own fork and allowing actions there. The checks here are going to fail due to permissions issues but if I see them green on your PR, I am happy to merge things.

However, there are other build failures (formatting I believe) that need to be taken care of!

File: src/index.ts (hunk 1 above)
Comment (vitalyal) [reply src/index.ts]: fixed

File: src/index.ts (hunk 2 above)
Comment (vitalyal) [reply src/index.ts]: fixed

**Comment by vitalyal** [comment]
Hi, build checks on my PR got green https://github.com/vitalyal/create-pull-request/pull/1/checks

**Comment by thomaseizinger** [comment]
> Hi, build checks on my PR got green [vitalyal/create-pull-request#1 (checks)](https://github.com/vitalyal/create-pull-request/pull/1/checks)

Thank you!
I forgot that I changed the workflows in a previous PR. Can you merge the PR on your fork and link me to the resulting workflow run that is going to be triggered? I'll see to properly document this.

**Comment by thomaseizinger** [comment]
> Hi @thomaseizinger , i had to fix new integration tests a bit and now the only failing test case is about reviewers because they are not configured as contributors in my fork https://github.com/vitalyal/create-pull-request/actions/runs/1055169889.
> As CONTRIBUTING.md states PR should be created from master, but it is not possible to update existing PR. do you want me to create another one?

Yeah that failing test is fine :)
Creating a new PR from your master branch would be superb, thank you for sticking with this and sorry about the testing mess! (I am open to suggestions in how to improve it.)

**Comment by thomaseizinger** [comment]
Can you also update the changelog? It is missing an `Unreleased` section at the top at the moment.

**Comment by thomaseizinger** [comment]
So actually, I realized this entire action might be pointless given the existence of the `gh` CLI. Do you have any experience using it inside GitHub workflows?

https://cli.github.com/manual/gh_pr_create

**Comment by vitalyal** [comment]
Ok, i've moved Ureleased section to top from the bottom of the file and assume link in that section will show more changes as long as they are merged to master.
As for gh CLI pr_create i do not have any experience with it. As i understand it is supposed to be used from local repo with changes to be pushed to PR. With your action i'm able to create PR from one remote branch to another. I also have other usecases where i need to create PR from local changes and i found https://github.com/marketplace/actions/create-pull-request action to be really useful as well

**Comment by vitalyal** [comment]
And here is PR from main to main https://github.com/thomaseizinger/create-pull-request/pull/61

**Comment by thomaseizinger** [comment]
> Ok, i've moved Ureleased section to top from the bottom of the file and assume link in that section will show more changes as long as they are merged to master.

Sorry, there was a misunderstanding there. The `[Unreleased]` part is actually just a link reference. It shouldn't be moved but instead should have a `## [Unreleased]` header at the top, similar to how it is done for `[1.0.0]`.

If you check the rendered output, this allows the heading to be clickable: https://github.com/thomaseizinger/create-pull-request/blob/master/CHANGELOG.md

> And here is PR from main to main #61

I am going to close this as a result.

**Pull request closed without merging**